_api_app/app/Sites/MultisitesMenuRenderService.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Sites;
+
+use App\Shared\Helpers;
+use App\Shared\Storage;
+use App\Sites\SitesDataService;
+
+class MultisitesMenuRenderService
+{
+    private $currentSite;
+    private $isEditMode;
+
+    /**
+     * Construct SitesRenderService instance
+     *
+     * @param string $currentSite
+     * @param bool $isEditMode
+     */
+
+    public function __construct(
+        $currentSite,
+        $isEditMode
+    ){
+        $this->currentSite = '';
+        $this->isEditMode = false;
+    }
+
+    public function getViewData()
+    {
+
+        $SitesDataService = new SitesDataService();
+        $multisites = $SitesDataService->get();
+        $data['multisite'] = [];
+        $i = 0;
+        foreach ($multisites as $sites) {
+            $isPublished = $this->isEditMode == true ? true : $sites['@attributes']['published'] == 1 ? true : false;
+            $isCurrentSite = $this->currentSite === $sites['name'] ? true : false;
+            $isAvailable = $this->isEditMode == true || $this->currentSite != $sites['name'] || ($sites['name'] == '' && $this->currentSite == '') ? true : false;
+            $displayName = $sites['title'] !== '' ? $sites['title'] : $sites["name"];
+
+            $data['multisite'] = $data['multisite'] + [
```

Comment:
Filter only required sites for the twig template here.
You can also add the `selected` class name here and make twig template clean and readble.